### PR TITLE
Centralise the checks for blob and form data support

### PR DESCRIPTION
**Pull request by matthew-andrews**
This helps with isomorphic-fetch (currently it's relying on a fork that has these extra few lines:- https://github.com/matthew-andrews/fetch/blob/master/fetch.js#L3-5) and (I think) makes fetch's implementation simpler.

### CODE REVIEW

File: fetch.js [hunk 1]
```diff
@@ -109,9 +110,9 @@
         this._bodyInit = body
         if (typeof body === 'string') {
           this._bodyText = body
-        } else if ('Blob' in self && Blob.prototype.isPrototypeOf(body)) {
+        } else if (blobSupport && Blob.prototype.isPrototypeOf(body)) {
```
Comment (matthew-andrews) [inline fetch.js]: wasn't sure about this because `blobSupport` actually checks a lot more than just `'Blob' in self`

**Comment by josh** [comment]
If it would help you more, I wouldn't mind centralizing all the checks to the top with a single object like

``` js
var support = {
  "Blob": howeverTestBlob,
  "FileReader": somethingTestFileReader
}
```

We could even export it as `fetch._support` if you wanted to runtime hack or inspect it.

**Comment by matthew-andrews** [comment]
Umm, I don't mind.  I would just like it to do its checks upfront so that I can [restore `self`](https://github.com/matthew-andrews/isomorphic-fetch/blob/master/server.js#L17) after `require`ing fetch.

I can definitely tidy them all up into a single object though.

**Comment by josh** [comment]
:heart:

**Comment by matthew-andrews** [comment]
Would you mind if I made a (minor) release for this & https://github.com/github/fetch/pull/77?

**Comment by josh** [comment]
no problem, done 115cd972a7d5dbc63bd2335a05976070e0ebc48c

**Pull request merged**